### DISCUSSION TITLE
See change log for underscore version changes

### DISCUSSION THREAD

**Discussion by spfennell** [Homebrew/discussions#1550] [Everyday usage]
Where can I find a changelog for underscore version changes? My software was working fine with llvm 12.0.0, but now with 12.0.0_1 it suddenly has a fatal error. As strange as this is, I have confirmed that this change to my system caused the error.

I want to understand:

- What exactly are underscore versions/formula changes for
- Can I find release notes/changelog on this formula update

**Comment by carlocab** [comment]
Try `brew log`.
```
❯ brew help log
Usage: brew log [options] [formula]

Show the git log for formula, or show the log for the Homebrew repository if
no formula is provided.

  -p, -u, --patch                  Also print patch from commit.
      --stat                       Also print diffstat from commit.
      --oneline                    Print only one line per commit.
  -1                               Print only one commit.
  -n, --max-count                  Print only a specified number of commits.
  -d, --debug                      Display any debugging information.
  -q, --quiet                      Make some output more quiet.
  -v, --verbose                    Make some output more verbose.
  -h, --help                       Show this message.
```

**Comment by carlocab** [comment]
Also, probably related to your `llvm` issue: https://github.com/Homebrew/homebrew-core/issues/78025

**Comment by SMillerDev** [comment]
> What exactly are underscore versions/formula changes for

Since this question was still outstanding, they're revisions of the formula. If just the formula changes but not the software version we can use a revision to get an update to users.
